### DISCUSSION TITLE
generate musicxml files in java

### DISCUSSION THREAD

**Discussion by aruns911** [w3c-cg/musicxml#451] [Q&A] [answered]
I'm looking for a java api that would generate musicxml files. I would like to pass in the music notes one at a time or as an array. Also would like to set key, clef, etc in the api. I want to run it in android. Can anyone recommend a a java api that I can use for this purpose?

**Comment by infojunkie** [comment]
[This Java library](https://github.com/Audiveris/proxymusic) could help you.

**Comment by mdgood** [comment, marked as answer]
Perhaps the [JFugue](http://www.jfugue.org/) library could help as well.
